src/components/AuthEvents/EditEvent.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import ReactDOM from 'react-dom';
+import SelectField from 'material-ui/SelectField';
+import MenuItem from 'material-ui/MenuItem';
+import DatePicker from 'material-ui/DatePicker';
+
+//components
+import Navigation from '../Navigation/Navigation';
+import Footer from '../Footer/Footer';
+import Category from '../category/GetCategories';
+
+import { getEventDetail } from '../../actions/eventActions';
+import { editEventAction } from '../../actions/authEvents.actions';
+
+
+const initialState = {
+	title: '',
+	location: '',
+	time: '',
+	date: '',
+	description: '',
+	image_url:'',
+	event_category: null
+};
+
+
+class EditEvent extends Component{
+
+    constructor(props){
+		super(props);
+        this.state = {
+			eventData: initialState,
+		};
+		// console.log('Constructor',editEvent);
+		this.handleInputChange = this.handleInputChange.bind(this);
+		this.handleCategoryChange = this.handleCategoryChange.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+
+	componentWillReceiveProps(nextProps){
+		this.setState({eventData: nextProps.editEvent.event})
+	}
+
+    componentWillMount = () => {
+		let eventId = this.props.match.params.id
+        this.props.getEventAction(eventId)
+    }
+
+    handleInputChange(e) {
+        const { eventData } = this.state;
+        eventData[e.target.name] = e.target.value;
+		this.setState({eventData});
+	}
+
+	handleCategoryChange(event, index, value) {
+		const { eventData } = this.state
+		eventData.event_category = value
+		this.setState({eventData})
+	}
+
+	setFechaDesde(x,event){
+        // console.log('initial, ',event);
+		// console.log(JSON.stringify(x));
+		const { eventData } = this.state
+		eventData.date = event
+		this.setState({eventData})
+    }
+
+	handleSubmit(e) {
+		e.preventDefault();
+		const event = this.state.eventData;
+		let eventId = this.props.match.params.id
+		console.log('handleSubmit', this.props);
+		this.props.editEventAction(eventId, event);
+		this.setState({eventData: initialState});
+		setTimeout(
+			function() {
+				this.props.history.push('/profile');
+			}
+				.bind(this), 1000);
```

Comment:
Binding can be avoided by using arrow functions.

Reply:
thank you, I'll take note and update accordingly